packages/web3/src/v3/WalletLink.ts

```diff
@@ -51,22 +60,32 @@ export class WalletLink {
         rootKeyNonce,
     }: {
         rootKey: ethers.Signer
-        walletAddress: string
+        walletAddress: Address
         rootKeyNonce: BigNumber
-    }) {
-        return rootKey.signMessage(packAddressWithNonce(walletAddress, rootKeyNonce))
+    }): Promise<string> {
+        const { domain, types, value } = createEip712LinkedWalletdData({
+            domain: this.eip712Domain,
+            nonce: rootKeyNonce,
+            wallet: walletAddress,
+        })
+        return this.signTypedData(rootKey, domain, types, value)
     }
 
-    private generateWalletSignatureForRootKey({
+    private async generateWalletSignatureForRootKey({
```

Comment:
why async?

Reply:
good eye. Had an await in this method before I refactored code. Removed.